src/recorder/jamrecorder.cpp

```diff
@@ -396,15 +396,30 @@ void CJamRecorder::Start() {
     // Ensure any previous cleaning up has been done.
     OnEnd();
 
+    QString error;
+
     // needs to be after OnEnd() as that also locks
     ChIdMutex.lock();
     {
-        currentSession = new CJamSession( recordBaseDir );
-        isRecording = true;
+        try
+        {
+            currentSession = new CJamSession( recordBaseDir );
+            isRecording = true;
+        }
+        catch ( const std::runtime_error& err )
+        {
+            currentSession = nullptr;
+            error = err.what();
+        }
     }
     ChIdMutex.unlock();
 
-    emit RecordingSessionStarted ( currentSession->SessionDir().path() );
+    if ( currentSession )
+    {
+        emit RecordingSessionStarted ( currentSession->SessionDir().path() );
+        return;
```

Comment:
I'd rather `else` emit the error, than `return` here, style-wise.

Reply:
I don't feel here it's an early return.  It's
```
void method() {
 // always
 if ( a ) {
  // something if a
 } else {
  // something else
 }
}
```
There's nothing after the end of the if/else before the method returns all by itself.  We're basically saying "either emit this or emit that" and idiomatically that's an if/else.

Reply:
> There's nothing after the end of the if/else before the method returns all by itself. We're basically saying "either emit this or emit that, done" and idiomatically that's an if/else.

I agree. It has little benefits right now. However, if it is ever extended, it would keep the code path simpler. This sounds like YAGNI, but at the same time we said in the referenced ticket, that it is hard to re-structure existing code that way without risk of breakage, so... if we want to get better there, we might want to start early.

Nevertheless, I've adjusted the code like you've suggested now (with the next push).

Reply:
The question is whether it was the right place for the return then.  If it was "return on failure", it should have been after the other emit.  It was more likely that the if block with the return (as the "positive" side) would be the one to grow.  This way, should any code get added to the positive side, then the if can be easily inverted and a return added.

Perhaps that's the better option now:
```
void method() {
 if ( !itWorked() ) {
  emit itFailed();
  return;
 }
 emit itDidNotFail();
 doMoreStuffMaybeOneDay();
}
```

Reply:
Yes, I think what you've suggested is the best version. I have adapted the code accordingly.